contracts/Client.sol

```diff
@@ -0,0 +1,222 @@
+    currentLoan = loanHistory[_id];
+
+    return (
+      _id, 
+      currentLoan.amount, 
+      currentLoan.startDate, 
+      currentLoan.endDate, 
+      currentLoan.interestRate, 
+      currentLoan.numberOfPayments, 
+      currentLoan.status
+    );
+  }
+
+  function Client(
+    address _clientWallet,
+    bytes32 _name,
+    bytes32 _homeAddress, 
+    bytes32 _birthday, 
+    uint256 _age, 
+    uint8 _gender, 
+    uint8 _education,
+    uint8 _householdSize,
+    uint8 _dependents,
+    bool _married,
+    uint256 _monthlyHouseholdIncome,
+    uint8 _numOfBusinesses,
+    uint8 _numOfEmployees,
+    uint256 _phoneNumber
+  ) {
+    clientInfo = PersonalInfo({
+      name: _name,
+      homeAddress: _homeAddress,
+      birthday: _birthday,
+      age: _age,
+      gender: _gender,
+      education: _education,
+      householdSize: _householdSize,
+      dependents: _dependents,
+      married: _married,
+      monthlyHouseholdIncome: _monthlyHouseholdIncome,
+      numOfBusinesses: _numOfBusinesses,
+      numOfEmployees: _numOfEmployees,
+      phoneNumber: _phoneNumber
+    });
+    clientWallet = _clientWallet;
+    numberOfJobs = 0;
+    numberOfDeposits = 0;
+  }
+
+  function getPersonalInfo() constant returns (
+    address clientWalletAddress,
+    bytes32 name, 
+    bytes32 homeAddress, 
+    bytes32 birthday, 
+    uint256 age, 
+    uint8 gender, 
+    uint8 education,
+    uint8 householdSize,
+    uint8 dependents,
+    bool married,
+    uint256 monthlyHouseholdIncome,
+    uint8 numOfBusinesses,
+    uint8 numOfEmployees,
+    uint256 phoneNumber
+  ) {
+    return ( 
+      clientWallet,
+      clientInfo.name,
+      clientInfo.homeAddress,
+      clientInfo.birthday,
+      clientInfo.age,
+      clientInfo.gender,
+      clientInfo.education,
+      clientInfo.householdSize,
+      clientInfo.dependents,
+      clientInfo.married,
+      clientInfo.monthlyHouseholdIncome,
+      clientInfo.numOfBusinesses,
+      clientInfo.numOfEmployees,
+      clientInfo.phoneNumber
+    );
+  }
+
+  struct Deposit{
+    uint256 amount;
+    uint256 datetime;
+    bool txType;  // true = deposit, false = withdrawal
+  }
+  uint256 public numberOfDeposits;
+  mapping ( uint256 => Deposit ) depositHistory;
+  modifier onlyBy(address _account) { require(msg.sender == _account); _; }
+
+  function deposit() 
+    payable 
+    onlyBy(clientWallet)
+    returns(bool) 
+    {
+      //auth
+      numberOfDeposits = numberOfDeposits + 1;
+      Deposit memory newDeposit = Deposit(msg.value, now, true);
+      depositHistory[numberOfDeposits] = newDeposit;
+      return true;
+  }
+
+  function getDeposit(uint256 _id) constant returns(uint256, uint256, bool) {
+    return (depositHistory[_id].amount, depositHistory[_id].datetime, depositHistory[_id].txType);
+  }
+
+  function withdrawFunds(uint256 _amount) 
+    onlyBy(clientWallet)
+    returns (bool) 
+  {
+    // require(_amount > this.balance); ?? throws an exception
```

Comment:
This throws because the `>` is reversed.  It should be `require(_amount <= this.balance); `